features/networking/ovn.feature

```diff
@@ -0,0 +1,145 @@
+    Then the step should succeed
+    """
+    And admin ensures "ovnkube-master" ds is deleted from the "openshift-ovn-kubernetes" project
+    And admin executes existing pods die with labels:
+      | app=ovnkube-master | 
+    Given I have a project
+    And I obtain test data file "networking/pod-for-ping.json"
+    When I run the :create client command with:
+      | f | pod-for-ping.json |
+    Then the step should succeed
+    #Now scale up CNO pod to 1 and check whether hello-pod is synced to NB db
+    Given I run the :scale admin command with:
+      | resource | deployment                 |
+      | name     | network-operator           |
+      | replicas | 1                          |
+      | n        | openshift-network-operator |
+    Then the step should succeed
+    #A minimum wait for 30 seconds is tested to reflect CNO deployment to be effective which will then re-spawn ovn pods
+    Given 30 seconds have passed
+    And admin waits for all pods in the "openshift-ovn-kubernetes" project to become ready up to 60 seconds
+    #Checking whether Kube API data is synced on OVN NB db which in this case is a test-pod created in earlier steps
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | hello-pod |
+  
+  # @author anusaxen@redhat.com
+  # @case_id OCP-30057
+  @admin
+  @destructive
+  Scenario: OVN DB should be updated correctly if a resource only exist in NB db but not in Kube API	
+    Given the env is using "OVNKubernetes" networkType
+    Given I have a project
+    And evaluation of `project.name` is stored in the :hello_pod_project clipboard
+    And I have a pod-for-ping in the project
+    #Checking whether Kube API data is synced on OVN NB db which in this case is a hello-pod created in earlier steps
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | hello-pod |
+    # Now scale down CNO pod to 0
+    Given admin uses the "openshift-network-operator" project
+    And I run the :scale admin command with:
+      | resource | deployment       |
+      | name     | network-operator |
+      | replicas | 0                |
+    Then the step should succeed
+    Given I register clean-up steps:
+    """
+    Given admin uses the "openshift-network-operator" project
+    And I run the :scale admin command with:
+      | resource | deployment       |
+      | name     | network-operator |
+      | replicas | 1                |
+    Then the step should succeed
+    """
+    And admin ensures "ovnkube-master" ds is deleted from the "openshift-ovn-kubernetes" project
+    And admin executes existing pods die with labels:
```

Comment:
@anuragthehatter this step doesn't sound right to me.

Reply:
@pruan-rht Existing step ```all existing pods die with labels``` doesn't support admin capabilities so added ```all | admin executes``` to allow flexibility to poke in system projects as well. I also had an option to switch to pseudo user and execute this non-admin'ly but guess the change will be def. useful in future. Also it won't break existing test which uses this step as its ```all | admin executes``` now.

Line 124 is deleting ds associated with ovnkube-master pods and 125 is making sure those all pods are gone successfully

Reply:
@pruan-rht Let me know if any further questions

Reply:
have to regressed it against existing scenarios that use the `all` existing pods die...`?

Reply:
Ya i did that initially and regressed again a case likehttps://github.com/openshift/verification-tests/blob/master/features/networking/multus.feature#L366. No issues 
Logs: https://privatebin-it-iso.int.open.paas.redhat.com/?13d0c61305f04132#u2ZwRH/SPHBSpqeCSHi/mlKB+JGPUQnGApuU0un2Fr0=